### ISSUE TITLE
Seems to eat the body

### ISSUE THREAD

**Issue by skandragon** [Cyprinus12138/otelgin#1]
Adding this to the middleware on my Gin server seems to make the request body become empty, and return EOF on read.

**Comment by skandragon** [comment]
I believe the defect is the wrapper trying to restore the body.

```
	// Restore the request body for further processing
	c.Request.Body = io.NopCloser(bytes.NewReader(body))
```

This works, but the existing code which wraps the (now empty) `c.Request.Body` in the `NopCloser` fails to pass any content along.

**Comment by Cyprinus12138** [comment]
hi @skandragon Thank you for raising the issue; I will check this problem soon.

**Comment by Cyprinus12138** [comment]
hi @skandragon the bug should have be fixed in the version v1.0.1, thanks again for your contribution!
[v1.0.1](https://github.com/Cyprinus12138/otelgin/releases/tag/v1.0.1)
